app/src/main/java/app/mywatchlist/MainActivity.kt

```diff
@@ -22,13 +22,13 @@ class MainActivity : ComponentActivity() {
             repeatOnLifecycle(Lifecycle.State.STARTED) {
                 homeViewModel.uiState.map { it.watchables }.distinctUntilChanged()
                     .collect { uiState ->
-                        println(uiState)
+                        println("UI State: " + uiState)
                     }
             }
         }
 
         setContent {
-            MainScreen()
+            MainScreen(homeViewModel)
```

Comment:
From the docs (https://developer.android.com/topic/libraries/architecture/viewmodel#best-practices):
> Don't pass ViewModels to other classes, functions or other UI components. Because the platform manages them, you should keep them as close to it as you can. Close to your Activity, fragment, or screen level composable function. This prevents lower level components from accessing more data and logic than they need.
```suggestion
            MainScreen()
```

Reply:
Edit: Apparently not true with dependency injection and the navigation library we are using (https://developer.android.com/jetpack/compose/libraries#hilt-navigation):
> call hiltViewModel() to get an instance of ExampleViewModel scoped to the destination as shown in the code snippet below